TOKEN_GUIDE.md

```diff
@@ -0,0 +1,132 @@
+
+## Setting up an Endpoint in Astro
+To create an endpoint in Astro, you need to add a Javascript or Typescript file to the `pages` directory. This file should have the file name, followed by the return datatype extension, followed by `.js` or `.ts`.The Javascript and Typescript extensions will be removed during the build process.
+
+The most common type of return data is JSON, so our file will be a `.json.ts` file. We also need to pass some parameters. In Astro, you do this using brackets for the file/folder name. 
+
+Our file path for generating tokens will be `pages/rtc/[channel]/[role]/[uid].json.ts`, which will result in the end point of `rtc/{channel}/{role}/{uid}.json`.
+
+## Define the endpoint
+Our endpoint will be a `GET` request because our objective is to retrieve a token. You can define this in Astro by exporting a `GET` function and returning a JSON Response.
+
+We also import the `agora-token` package and bring in our `APP_ID` and `APP_CERTIFICATE`, which we defined in the `.env` file.
+
+
+```ts
+import agoraToken from "agora-token";
+import type { APIContext } from "astro";
+
+const APP_ID = import.meta.env.APP_ID;
+const APP_CERTIFICATE = import.meta.env.APP_CERTIFICATE;
+
+export async function GET({ params }: APIContext) {
+  return new Response(
+    JSON.stringify({
+      rtcToken: "token",
+    })
+  )
+}
+```
+
+## CORS
+We first need to set up our headers so that we don't encounter any CORS issues when using our backend.
+
+```ts
+export async function GET({ params }: APIContext) {
+    const headers = new Headers({
+        "Access-Control-Allow-Origin": "*",
+        "Access-Control-Allow-Methods": "GET, OPTIONS",
+    });
+
+    return new Response(JSON.stringify({
+        rtcToken: "token"
+    }), { headers })
+}
+```
+
+* "Access-Control-Allow-Origin": "*" allows all domains to access the server, which is helpful for APIs or during development.
+* "Access-Control-Allow-Methods": "GET, OPTIONS" specifies that only GET and OPTIONS methods are permitted, controlling how other sites can interact with the server.
+
+These headers will allow our API to work well in production and development, but for production you should limit only the origins that should be allowed.
+
+## Input Checks
+To generate the token, we need a few input parameters that we retrieve from the route. We need to check that `channel` and `uid` are not empty. If they are, return a Bad Request Error.
+
+The role determines what type of access the user requesting a token should have. There are two possible roles:
+* publisher - Can publish (or send) their data to the channel. They can also receive data from the channel.
+* subscriber - Can only receive data and cannot publish data.
+
+Those are the only two valid inputs. If one of those inputs is in the URL, we can assign the `role` variable to a number predefined within the `agora-token` package. If the request does not include one of these options, we send back a Bad Request Error.
```

Comment:
The second sentence is a bit vague. In the code it shows an enum but in the explanation it says "number"
> If one of those inputs is in the URL, we can assign the `role` variable to a number predefined within the `agora-token` package.

Reply:
I looked into the code and saw it returned a number, and I though using "number" would simplify the guide. However, I think that's unnecessary, I changed to to `enum`

Reply:
The number part feels like it comes out of no where. maybe explain that you're using an enumeration to use human readable names that align with the numerical values expected by Agora.

Reply:
I just the "enum". I feel like that does enough of the work. These are developers we're writing to. They should know what an enum is and explaining it might be going too deep for no reason

> If one of those inputs is in the URL, we can assign the `role` variable to an enum predefined within the `agora-token` package.